### PR TITLE
C973 Use full CRUD for postgres access

### PR DESCRIPTION
**Pull request by BSick7**
This updates the module to treat database schema changes as CRUD objects instead of "ensure on plan".

### CODE REVIEW

File: nullstone.tf [hunk 1]
```diff
@@ -3,6 +3,9 @@ terraform {
     ns = {
       source = "nullstone-io/ns"
     }
+    restapi = {
```
Comment (ssickles) [inline nullstone.tf:6]: Does it make sense to call it "restapi"? Seems like we would call it something like "db_admin". If we decided to follow this pattern for something else, I would imagine we would create a new provider with a different name.

**Review by ssickles — APPROVED** [review]
This looks great, just one comment about the naming. Seems odd to name it based on "how it does it" vs "what it does".

File: nullstone.tf (hunk 1 above)
Comment (BSick7) [reply nullstone.tf:6]: That's the name of the Terraform provider.

This line creates an alias named `restapi` to the provider and while it's possible to name it anything, it feels a bit wrong to do that in Terraform -- TF contributors use the resource prefix `restapi_` to search for docs on registry.terraform.io.

**Review by ssickles — APPROVED** [review]

File: create.tf [hunk 2]
```diff
@@ -1,40 +1,110 @@
...
-  depends_on = [data.aws_lambda_invocation.create-user]
-}
+provider "restapi" {
+  uri                  = coalesce(local.db_admin_func_url, "https://noop")
+  write_returns_object = true
+
+  aws_v4_signing {
+    service           = "lambda"
+    access_key_id     = try(local.db_admin_invoker["access_key"], null)
+    secret_access_key = try(local.db_admin_invoker["secret_key"], null)
```
Comment (ssickles) [inline create.tf:8]: How does this key get passed? Is this secure?

File: create.tf [hunk 3]
```diff
@@ -1,40 +1,110 @@
...
-    type = "create-db-access"
-    metadata = {
-      databaseName = local.database_name
-      username     = local.username
-    }
-  })
-
-  depends_on = [data.aws_lambda_invocation.create-user]
-}
+provider "restapi" {
```
Comment (ssickles) [inline create.tf:1]: OK, this entire file is cool as hell. It is very clear what is going on here. Well done sir!

**Review by ssickles — APPROVED** [review]
Looks pretty straightforward, I know the bulk of the work is in the rest_api provider.

File: create.tf (hunk 2 above)
Comment (BSick7) [reply create.tf:8]: This gets passed by state file outputs. It's secure as long as we protect the state files.

**Pull request merged**